src/signalfx_metadata.py

```diff
@@ -0,0 +1,255 @@
+                    meta={'0': True},
+                    type=val[1].lower(),
+                    type_instance=metric,
+                    values=[val[0]]).dispatch()
+
+
+def putnotif(property_name, message):
+    notif = collectd.Notification(plugin=PLUGIN_NAME,
+                                  plugin_instance=property_name,
+                                  type_instance=TYPE_INSTANCE,
+                                  type=TYPE)
+    notif.severity = 4  # OKAY
+    notif.message = message
+    notif.dispatch()
+
+
+def plugin_config(conf):
+    for kv in conf.children:
+        if kv.key == 'Notifications':
+            if kv.values[0]:
+                collectd.register_notification(receive_notifications)
+        elif kv.key == 'URL':
+            global POST_URL
+            POST_URL = kv.values[0]
+        elif kv.key == 'Token':
+            global API_TOKEN
+            API_TOKEN = kv.values[0]
+        elif kv.key == 'Timeout':
+            global TIMEOUT
+            TIMEOUT = int(kv.values[0])
+        elif kv.key == 'NotifyLevel':
+            global NOTIFY_LEVEL
+            if string.lower(kv.values[0]) == "okay":
+                NOTIFY_LEVEL = 4
+            elif string.lower(kv.values[0]) == "warning":
+                NOTIFY_LEVEL = 2
+            elif string.lower(kv.values[0]) == "failure":
+                NOTIFY_LEVEL = 1
+        else:
+            raise Exception("unknown config parameter '%s'" % kv.key)
+
+
+def send():
+    send_notifications()
+
+
+def write_notifications(host_info):
+    for property_name, property_value in host_info.iteritems():
+        putnotif(property_name, property_value)
+
+
+# source http://code.activestate.com/recipes/439093-get-names-of-all-up-network-interfaces-linux-only/#c7
+# trying not to bundle too much code along with this but this was the best, most portable version of what
+# i needed
+def all_interfaces():
+    is_64bits = sys.maxsize > 2 ** 32
+    struct_size = 40 if is_64bits else 32
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    max_possible = 8  # initial value
+    while True:
+        bytes = max_possible * struct_size
+        names = array.array('B', '\0' * bytes)
+        outbytes = struct.unpack('iL', fcntl.ioctl(
+            s.fileno(),
+            0x8912,  # SIOCGIFCONF
+            struct.pack('iL', bytes, names.buffer_info()[0])
+        ))[0]
+        if outbytes == bytes:
+            max_possible *= 2
+        else:
+            break
+    namestr = names.tostring()
+    return [(namestr[i:i + 16].split('\0', 1)[0],
+             socket.inet_ntoa(namestr[i + 20:i + 24]))
+            for i in range(0, outbytes, struct_size)]
+
+
+def get_interfaces(host_info={}):
+    for interface, ipaddress in all_interfaces():
+        if ipaddress == "127.0.0.1":
+            continue
+        host_info["ipaddress_" + interface] = ipaddress
+        host_info["fqdn_" + interface] = socket.gethostbyaddr(ipaddress)[0]
```

Comment:
The gethostbyaddr call is obsolete as far a syscalls go.

http://linux.die.net/man/3/gethostbyname

Should you use getaddrinfo instead?